apps/settings/js/datasync/history_adapter.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+/* globals IAC */
+/* exported SynctoHistoryAdapter */
+
+var HistoryAPI = {
+  addPlaces: function(places) {
+    return IAC.request('sync-history', {
+      method: 'addPlaces',
+      args: [places]
+    });
+  },
+
+  addPlace: function(place) {
+    return IAC.request('sync-history', {
+      method: 'addPlace',
+      args: [place]
+    });
+  }
+};
+
+var SynctoHistoryAdapter = {
+  update: function(kintoCollection) {
+    function updateHistoryCollection(list) {
+      var historyRecords = list.data;
+      var places = [];
+      historyRecords.forEach(function (decryptedRecord){
+        var record = decryptedRecord.payload;
+        if(!record.histUri || !record.visits || !record.visits[0]){
+          return ;
+        }
+
+        var visits = [];
+        record.visits.forEach(function (elem){
+          visits.push(elem.date);
+        });
+
+        var place = {
+          url: record.histUri,
+          title: record.title,
+          visits: visits,
+          last_modified: decryptedRecord.last_modified
+        };
+        places.push(place);
+      });
+
+      return HistoryAPI.addPlaces(places);
+    }
+
+    return kintoCollection.list().then(list => {
```

Comment:
In any case, this file mixes use of `.then(function() {` with use of `.then(() => {` which is not consistent. Is there a style guide for gaia about whether we should be using arrow functions yet in new code or not? I personally like arrow functions a lot because they automatically bind `this` to the parent function, which saves me a lot of bugs. :)